ihatemoney/web.py

```diff
@@ -640,7 +642,60 @@ def change_lang(lang):
 def settle_bill():
     """Compute the sum each one have to pay to each other and display it"""
     bills = g.project.get_transactions_to_settle_bill()
-    return render_template("settle_bills.html", bills=bills, current_view="settle_bill")
+    bill_forms = [get_payment_form(bill) for bill in bills]
+
+    return render_template(
+        "settle_bills.html",
+        bills=bills,
+        bill_forms=bill_forms,
+        current_view="settle_bill"
+    )
+
+
+def get_payment_form(settlement_bill, **kwargs):
+    form = get_billform_for(g.project, set_default=False, bill_type=BillTypes.settlement.value)
+    bill = Bill()
```

Comment:
You probably can pass all the parameters to the `Bill` init directly.

Reply:
Or, what about having :+1: 
1. a`from_settlement` class method for the `Bill`, which creates the bill;
2. a function which takes this bill to fill the form.